From jic23@cam.ac.uk  Thu Apr 22 16:37:59 2010
From: Jonathan Cameron <jic23@cam.ac.uk>
Date: Thu, 11 Mar 2010 17:29:48 +0000
Subject: staging: iio: lis3l02dq - incorrect ws used in container of call.
To: Greg KH <greg@kroah.com>,  "linux-iio@vger.kernel.org" <linux-iio@vger.kernel.org>
Message-ID: <4B99288C.3080601@cam.ac.uk>

The word oops comes to mind.  Original patch to merge the two work queues
in here (prior to Greg taking them into staging) changed the top half to
only use one of them and the bottom half to assume it was the other.

Currently causes a NULL pointer dereference if you enable any of the events
on an lis3l02dq. Just goes to show I need a few more regression tests.


Signed-of-by: Jonathan Cameron <jic23@cam.ac.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/iio/accel/lis3l02dq_core.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/iio/accel/lis3l02dq_core.c
+++ b/drivers/staging/iio/accel/lis3l02dq_core.c
@@ -618,7 +618,7 @@ static int lis3l02dq_thresh_handler_th(s
 static void lis3l02dq_thresh_handler_bh_no_check(struct work_struct *work_s)
 {
 	struct iio_work_cont *wc
-		= container_of(work_s, struct iio_work_cont, ws_nocheck);
+		= container_of(work_s, struct iio_work_cont, ws);
 	struct lis3l02dq_state *st = wc->st;
 	u8 t;
 
